src/sqlfluff/linter.py

```diff
@@ -808,11 +860,26 @@ def paths_from_path(self, path, ignore_file_name='.sqlfluffignore', ignore_non_e
                     if fname.endswith(ext):
                         buffer.append(fpath)
 
+        if not ignore_files:
+            return sorted(buffer)
+
         # Check the buffer for ignore items and normalise the rest.
         filtered_buffer = []
+
         for fpath in buffer:
             if fpath not in ignore_set:
                 filtered_buffer.append(os.path.normpath(fpath))
+            elif is_exact_file:
+                print(
+                    "WARNING: Exact file path %s was given but "
+                    "it was ignored by a %s pattern, "
+                    "re-run with `--not-ignore-files` to "
+                    "skip %s" % (
+                        path,
+                        ignore_file_name,
+                        ignore_file_name,
+                    )
+                )
```

Comment:
Can we use `linter_logger.warning()` here rather than `print`? I think that will be more future compatible.

Reply:
The reason I'm using `print` instead of `linter_logger.warning()` is that I wanted to give feedback to the user. I believe we are doing the same here: https://github.com/sqlfluff/sqlfluff/blob/master/src/sqlfluff/linter.py#L270

By default `linter_logger.warning()` wasn't showing up, should we change that default instead?

Reply:
Good catch. I've just addressed that remaining print statement this in #497. [I've also made warnings red by default too, so they're more obvious).

The warnings *should* come through regardless - can you check that it actually the case though? I've tested on my machine and they are coming through.

Reply:
Tested now and it's working!